### PR TITLE
Fix gradient checkpointing bug in trocr

### PR DESCRIPTION
**Pull request by KMFODA**
This PR fixes a bug that a user can encounter while using generate and models that use gradient_checkpointing.

Fixes Issue https://github.com/huggingface/transformers/issues/21737

cc @younesbelkada or @gante

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this!

File: src/transformers/models/trocr/modeling_trocr.py [hunk 1]
```diff
@@ -664,6 +664,13 @@ def forward(
             # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
             encoder_attention_mask = _expand_mask(encoder_attention_mask, inputs_embeds.dtype, tgt_len=input_shape[-1])
 
+        if self.gradient_checkpointing and self.training:
+            if use_cache:
+                logger.warning_once(
+                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache =" " False`..."
```
Comment (younesbelkada) [inline src/transformers/models/trocr/modeling_trocr.py:670]:
```suggestion
                    "`use_cache = True` is incompatible with gradient checkpointing. Setting `use_cache = False`..."
```

**Review by amyeroberts — APPROVED** [review]
Thanks for the fix (and all of the others too!)
